src/uwtools/config.py

```diff
@@ -752,3 +753,17 @@ def create_config_obj(
                     raise ValueError(err_msg)
                 # Dump to file:
                 dump_method(path=outfile, cfg=config_obj)
+
+
+def export_variables(config_dict: dict, section_path: list, log: Logger) -> None:
+    """
+    Allows users to export sections of the config file as environment variables.
```

Comment:
I think using the word "export" here could be misleading, as "to export an environment variable" has a specific conventional meaning, and that's not what's happening here. I think we could just state explicitly what we _are_ doing, e.g.
> Descend into the config via the given section keys, then print the contents of the located subtree as key=value pairs, one per line.
Or something like that.